src/java.desktop/share/classes/com/sun/imageio/plugins/tiff/TIFFIFD.java

```diff
@@ -56,12 +56,12 @@
     // A set of tag numbers corresponding to tags essential to decoding
     // the image and metadata required to interpret its samples.
     //
-    private static volatile Set<Integer> essentialTags = null;
+    private static volatile Set<Integer> essentialTags;
 
     private static void initializeEssentialTags() {
         Set<Integer> tags = essentialTags;
         if (tags == null) {
-            essentialTags = tags = Set.of(
+            essentialTags = Set.of(
```

Comment:
What is the purpose of the local tags here?

Reply:
Looks like there's no purpose, `tags` is not used after assignment

Reply:
Maybe it's kind of a protection from a race. Yet it's possible either way: another thread could see `essentialTags == null` before this one initialises the field.

Reply:
I think we can drop it completely.

Reply:
Agree.

Reply:
I still suggest to remove the local tag completely.